src/types/module.rs

```diff
@@ -121,7 +121,12 @@ impl PyModule {
     /// May fail if the module does not have a `__file__` attribute.
     #[cfg(not(all(windows, PyPy)))]
     pub fn filename(&self) -> PyResult<&str> {
-        unsafe { self.str_from_ptr(ffi::PyModule_GetFilename(self.as_ptr())) }
+        unsafe {
+            self.str_from_ptr(
+                #[allow(deprecated)]
+                ffi::PyModule_GetFilename(self.as_ptr()),
```

Comment:
I think we should change this to use `PyModule_GetFilenameObject`.

So far this series of PRs has only been the minimum changes for #1289, but maybe an update here would be appropriate?

Reply:
I agree we can change this. It looks like `PyModule_GetFileameObject` is guaranteed to return `PyString` on success, or `NULL` on failure, so you should be able to do something like `self.py().from_owned_ptr_or_err::<PyString>(PyModule_GetFilenameObject(self.as_ptr()))?.to_str()`